File: docs/snippets/all/archetypes/asset3d_simple.cpp
// Log a simple 3D asset.

#include <rerun.hpp>

#include <iostream>

int main(int argc, char* argv[]) {
    if (argc < 2) {
        std::cerr << "Usage: " << argv[0] << " <path_to_asset.[gltf|glb|obj|stl]>" << std::endl;
        return 1;
    }

    const auto path = argv[1];

    const auto rec = rerun::RecordingStream("rerun_example_asset3d");
    rec.spawn().exit_on_failure();

    rec.log_static("world", rerun::ViewCoordinates::RIGHT_HAND_Z_UP); // Set an up-axis
    rec.log("world/asset", rerun::Asset3D::from_file_path(path).value_or_throw());
}

File: docs/snippets/all/archetypes/asset3d_simple.rs
//! Log a simple 3D asset.

use rerun::external::anyhow;

fn main() -> anyhow::Result<()> {
    let args = std::env::args().collect::<Vec<_>>();
    let Some(path) = args.get(1) else {
        anyhow::bail!("Usage: {} <path_to_asset.[gltf|glb|obj|stl]>", args[0]);
    };

    let rec = rerun::RecordingStreamBuilder::new("rerun_example_asset3d").spawn()?;

    rec.log_static("world", &rerun::ViewCoordinates::RIGHT_HAND_Z_UP())?; // Set an up-axis
    rec.log("world/asset", &rerun::Asset3D::from_file_path(path)?)?;

    Ok(())
}
